cluster-autoscaler/proposals/cp_agnostic_ca.md

```diff
@@ -0,0 +1,158 @@
+# Decoupling ClusterAutoScalerCore (separate library) from CloudProvider implementations
+##### Author: bhargav-naik
+
+### Introduction
+Current ClusterAutoscaler is released as a bundle which includes core and all the CloudProviders included.
+When someone intends to write implementation of the CloudProvider they need to fork the intended version ClusterAutoscaler and provide their implementation.
+In this proposal, a different design approach is presented which will decouple the ClusterAutoscaler core from CloudProviders and make ClusterAutoScalerCore available as a separate release. 
+
+### Issues
+- In the current design the ClusterAutoscaler core has dependency on the CloudProvider implementations.
+- For proprietary CloudProvider implementations it results in an development cycle to fork new version from public repo and port the CloudProvider implementation to the latest fork.
+- Also the ClusterAutoscaler library comes with a lot of unnecessary CloudProvider specific dependencies which might not be required for other CloudProviders. (We dont want to address this)
```

Comment:
Why to add something that won't be addressed as part of this proposal.

Reply:
Removed.